### PR TITLE
[Chart] Add option to set loadBalancerSourceRanges in webserver service

### LINKED ISSUE #17660: Chart - Add option to set loadBalancerSourceRanges in webserver service
**Description**
It is not possible to create the service type "LoadBalancer" with option loadBalancerSourceRanges

**Use case / motivation**
When created webserver service with type "LoadBalancer" this value is "0.0.0.0/0" per default, In order to limit which client IP's can access the  Load Balancer, we can specify loadBalancerSourceRanges. 

**Are you willing to submit a PR?**
https://github.com/apache/airflow/pull/17666

### PR DESCRIPTION
**Pull request by Diogo-Costa**
Description
It is not possible to create the service type "LoadBalancer" with option loadBalancerSourceRanges

Use case / motivation
When created webserver service with type "LoadBalancer" this value is "0.0.0.0/0" per default, In order to limit which client IP's can access the Load Balancer, we can specify loadBalancerSourceRanges.

related: #17660
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: chart/templates/webserver/webserver-service.yaml [hunk 1]
```diff
@@ -51,3 +51,9 @@ spec:
   {{- if .Values.webserver.service.loadBalancerIP }}
   loadBalancerIP: {{ .Values.webserver.service.loadBalancerIP }}
   {{- end }}
+  {{- if .Values.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges:
+  {{- range $cidr := .Values.service.loadBalancerSourceRanges }}
+    - {{ $cidr }}
+  {{- end }}
```
Comment (jedcunningham) [inline chart/templates/webserver/webserver-service.yaml]:
```suggestion
  {{- toYaml .Values.service.loadBalancerSourceRanges | nindent 4 }}
```

I think this is cleaner?

File: chart/values.schema.json [hunk 2]
```diff
@@ -1818,6 +1818,12 @@
                             ],
                             "default": null
                         }
+                        },
+                        "loadBalancerSourceRanges": {
+                            "description": "Webserver Service loadBalancerSourceRanges.",
+                            "type": "array",
+                            "default": []
```
Comment (jedcunningham) [inline chart/values.schema.json]:
```suggestion
                            "default": [],
                            "examples": [
                                "143.231.0.0/16"
                            ]
```

**Review by jedcunningham — CHANGES_REQUESTED** [review]
Can you add tests in `tests/test_webserver.py::WebserverServiceTest`?

Also missing this feature for flower. Do you want to add that also?

File: chart/values.yaml [hunk 3]
```diff
@@ -701,6 +701,10 @@ webserver:
     #     port: 80
     #     targetPort: 8888
     loadBalancerIP: ~
+    ## Limit load balancer source ips to list of CIDRs
+    # loadBalancerSourceRanges:
+    #   - "143.231.0.0/16"
```
Comment (jedcunningham) [inline chart/values.yaml]:
```suggestion
    #   - "10.123.0.0/16"
```

Thoughts on the examples being a private block instead?

Comment (Diogo-Costa) [reply chart/values.yaml]: I put it only at example, fixing this in next commit

**Comment by jedcunningham** [comment]
Looks like static checks are failing. I'd also suggest that you [set up the pre-commit hooks](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst#prerequisites-for-pre-commit-hooks), they help catch these issues early 👍.

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @Diogo-Costa for the contribution, and congrats on your first commit!

**Pull request merged**